packages/client/lib/config.ts

```diff
@@ -184,15 +184,25 @@ export class Config {
   }
 
   /**
-   * Returns the directory for storing the client sync data
+   * Returns the directory for storing the client chain data
    * based on syncmode and selected chain (subdirectory of 'datadir')
    */
-  getSyncDataDirectory(): string {
-    const syncDirName = this.syncmode === 'light' ? 'lightchaindata' : 'chaindata'
-    const chain = this.common.chainName()
-    const networkDirName = chain === 'mainnet' ? '' : `${chain}/`
+  getChainDataDirectory(): string {
+    const networkDirName = this.common.chainName()
+    const chainDataDirName = this.syncmode === 'light' ? 'lightchain' : 'chain'
```

Comment:
In principle, the data which light/full reports should be the same, right? Only light will report a subset of the full data (IIRC Light only downloads Block Headers and not Block Bodies / state - right? Block Headers can also be used in full sync again, don't have to re-request those).

Reply:
Haven't spent that much time thinking about `lightsync` either 😋 , since we don't have a focus on this right now. On first thought I would assume that mixing this in one folder is not what one would want. If you start with a full sync to Block x, then continue to Block y with a light sync, all the block bodies from x to y would be missing? 🤔

Anyhow. This is really a simple change if we would want to combine, nothing which would need to be decided now, I just took this over "as is" from the existing implementation.

Reply:
Maybe you are right - it is not a problem for this PR anyways, just wanted to mention it 😄